packages/core/src/credential/Credential.ts

```diff
@@ -396,13 +396,17 @@ export async function verify(
     if (!isSchemaValid) throw new SDKErrors.CredentialUnverifiableError()
   }
 
-  if (challenge || !allowUnsigned) {
-    const isSignatureCorrect = await verifySignature(credential, {
+  let isSignatureCorrect = false
+  if (challenge || credential.claimerSignature) {
+    isSignatureCorrect = await verifySignature(credential, {
       challenge,
       resolver,
     })
-    if (!isSignatureCorrect) throw new SDKErrors.SignatureUnverifiableError()
+  } else if (allowUnsigned) {
+    isSignatureCorrect = true
```

Comment:
the logic is sound. just wondering if we shouldn't throw a different error here - which would also allow dropping the `isSignatureCorrect` variable that does not necessarily make it more readable

Reply:
[dd87752](https://github.com/KILTprotocol/sdk-js/pull/587/commits/dd877524199c8777cc81790892ece0ec62af2942)